modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -329,20 +329,50 @@ object SeqexecEngine {
     (configStatus ++ pendingConfig).toList.sortBy(_._1)
   }
 
-  def pendingConfigStatus(executions: List[List[engine.Action \/ engine.Result]]): List[(Resource, ActionStatus)] =
+  /**
+   * Calculates the config status for pending steps
+   */
+  private[server] def pendingConfigStatus(executions: List[List[engine.Action \/ engine.Result]]): List[(Resource, ActionStatus)] =
     executions.map {
-      s => s.separate.bimap(_.map(_.kind).flatMap(kindToResult), _.map(_.kind).flatMap(kindToResult))
+      s => s.separate.bimap(_.map(_.kind).flatMap(kindToResource), _.map(_.kind).flatMap(kindToResource))
     }.flatMap {
       x => x._1 ::: x._2
     }.distinct.strengthR(ActionStatus.Pending).sortBy(_._1)
 
-  def stepConfigStatus(step: StepAR): List[(Resource, ActionStatus)] =
+  /**
+   * Overall pending status for a step
+   */
+  private def stepConfigStatus(step: StepAR): List[(Resource, ActionStatus)] =
     engine.Step.status(step) match {
       case StepState.Pending => pendingConfigStatus(step.executions)
       case _                 => configStatus(step.executions)
     }
 
+  protected[server] def observeStatus(executions: List[List[engine.Action \/ engine.Result]], configStatus: List[(Resource, ActionStatus)]): ActionStatus = {
+    def containsPartial(e: List[engine.Action \/ engine.Result]): Boolean =
```

Comment:
This is a special case. A `Partial` is a `Result` but we should make the state of the Observation as `Running`
@jluhrs Do you think this is right. It is a bit weird that `Partial` is a `Result`

Reply:
I agree that it is weird. I had a similar problem recently with this.